search/includes/classes/class-health.php

```diff
@@ -315,6 +360,33 @@ public static function filter_expected_post_rows( $rows, $post_types, $post_stat
 		return $filtered;
 	}
 
+	public static function simplified_diff_document_and_prepared_document( $document, $prepared_document ) {
+		$ignored_keys = array(
+			// This field is proving problematic to reliably diff due to differences in the filters
+			// that run during normal indexing and this validator
+			'post_content_filtered',
+
+			// Meta fields from EP's "magic" formatting, which is non-deterministic and impossible to validate
+			'datetime',
+			'date',
+			'time',
+		);
+
+		foreach ( $document as $key => $value ) {
+			if ( in_array( $key, $ignored_keys, true ) ) {
+				continue;
+			}
+
+			if ( is_array( $value ) ) {
+				$recursive_diff = self::diff_document_and_prepared_document( $document[ $key ], $prepared_document[ $key ] );
```

Comment:
Should this be calling `simplified_diff_document_and_prepared_document()` instead?